scripts/fix_keen_preprint_keys.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import sys
+import django
+from django.db import transaction
+django.setup()
+
+from osf.models import Node
+from scripts import utils as script_utils
+from website import settings
+from website.app import init_app
+
+from keen import scoped_keys
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+def valid_keen_key(keenio_read_key, node_id):
+    try:
+        ret = scoped_keys.decrypt(settings.KEEN['public']['master_key'], keenio_read_key)
+        return ret['filters'][0]['property_value'] == node_id
+    except Exception as error:
+        logger.info('Error on {}:'.format(node_id))
+        logger.exception(error)
+        return True
```

Comment:
False will reroll the key -- if it can't decrypt the key in the first place it should be investigated outside of the context of this ticket. The only instance I ran into was a preprint that was created prior to setting up keen. It shouldn't happen on staging/prod, but wanted to add a check to make sure the script doesn't fail.